### DISCUSSION TITLE
Help converting production assessment to ss3sim model

### DISCUSSION THREAD

**Discussion by okenk** [ss3sim/ss3sim#415] [Q&A]
Hi,

I am working on converting [this assessment](https://github.com/pfmc-assessments/petrale/tree/main/models/2023.a034.001) to an ss3sim simulation model. I am most of the way there thanks to [the vignette](https://ss3sim.github.io/ss3sim/articles/making-models.html) (thank you!), but had some questions:

1. Starter file modification item 8. I cannot find `F_report_basis` in the starter file or in the r4ss starter list. Is that advice for a different version of ss3?
2. Control file modification item 1. I am a bit confused what this is referring to, as I cannot find `dev_adjust_method` in the control file.
3. What is the difference between `do_recdev` options 1 and 2?
4. `max_bias_adj` should default to zero for population simulations, correct? That is generally an estimation issue? (Just clarifying that I interpreted the directions correctly.)
5. Mean body weight: if the production assessment does use these data for discard rates, is there a good workaround you would recommend?

Thank you!!!

**Comment by iantaylor-NOAA** [comment]
It's great to hear of someone adding a new model to ss3sim. I think increasing our ability to do so will make the package all the more useful in the future. 

Here are a few answers. If the mismatch in names is a source of confusion, perhaps a pull request for changes to the vignette makes sense.

1. SS3 now calls this `F_std_basis` instead of `F_report_basis`, but it is still returned as `$F_report_basis` by `r4ss::SS_read()`. The line is here: https://github.com/pfmc-assessments/petrale/blob/main/models/2023.a034.001/starter.ss#L43.
2. I'm assuing `dev_adjust_method` is referring to this line: https://github.com/pfmc-assessments/petrale/blob/main/models/2023.a034.001/petrale_control.ss#L29 which r4ss calls $time_vary_adjust_method.
3. do_recdev option 1 is zero-centered deviations and option 2 is simple-deviations as described here: https://nmfs-ost.github.io/ss3-doc/SS330_User_Manual_release.html#recruitment-deviation-setup. Over the past few years there's been a shift to recommending option 2 (discussed last week at the NWFSC assessment team meeting).
4. The treatment of `max_bias_adj` in population simulations, depends on the context. It should definitely not have the ramp used in an estimation model (which deals with heterogeneous variability due to variable info in the data about the devs), but my memory is that ss3sim applies an adjustment to the simulated devs before entering them into SS3 so `max_bias_adj` should be 0. If that adjustment is not applied, then I think `max_bias_adj` should be 1.0 for all years. I would trust the ss3sim documentation on this.
5. In the petrale sole assessment, the mean body weight of the discards is somewhat redundant with the discard rates, so I think could be removed with almost zero impact on the model results. I would just remove that section of the data file and rerun the model to confirm that there are no large changes. Depending on your goals for the ss3sim analysis, you could further simplify the model by removing the retention function and discard rate data as well entirely because the discard rates are very low. The model would run faster with fewer parameters and fewer data but it would take more work to make those changes.

**Comment by kellijohnson-NOAA** [comment]
Thanks for documenting the trouble that you had @okenk and thanks @iantaylor-NOAA for providing such detailed help. I am now in the process of creating a function that will do a lot of this work for you. There are still a few things that will have to be done manually but the vignette will be updated for those. So, when you go to convert the sablefish and other models things should be easier and can be done with a function (not yet pushed) called `create_om()` and then `create_em()` can be used on the results of the former.
Thank you for your patience.
